voltdbclient/url.go

```diff
@@ -0,0 +1,33 @@
+package voltdbclient
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+)
+
+var errBadURL = errors.New("voltdb: bad connection url")
+
+func parseURL(src string) (*url.URL, error) {
+	u, err := url.Parse(src)
+	if err != nil {
+		return nil, err
+	}
+	if u.User == nil {
+		u.User = &url.Userinfo{}
+	}
+	if u.Scheme != "voltdb" {
+		if u.Host == "" {
+			if u.Opaque != "" {
+				u.Host = fmt.Sprintf("%s:%s", u.Scheme, u.Opaque)
+			} else {
+				u.Host = fmt.Sprintf("%s:%s", u.Scheme, "21212")
+			}
+		}
+	} else {
```

Comment:
If  `u.Host==""` is treated as localhost, opening a tcp connection on `:21212` is allowed and the same as opening tcp connection on `localhost:21212`